core/lz4/CMakeLists.txt

```diff
@@ -8,7 +8,23 @@ find_package(LZ4 REQUIRED)
 find_package(xxHash REQUIRED)
 
 target_sources(Core PRIVATE src/ZipLZ4.cxx)
-target_link_libraries(Core PRIVATE xxHash::xxHash LZ4::LZ4)
+# In order to avoid to link libxxHash.a twice to libCore in the final
+# link command to create libCore, we distinguish 3 cases:
+# 1. builtin_lz4 and system xxhash
+#    -> Link lz4 to libCore and make its headers location known
+# 2. builtin_lz4 and builtin_xxhash
+#    -> like 1., but make the location of xxhash known. It's not needed
+#       to explicitly link xxhash since it is linked to lz4
+# 3. system lz4 and system xxhash
+#     -> make the headers and libraries locations known
```

Comment:
Is the case "system lz4 and builtin_xxhash" impossible?

Reply:
it's possible and works well in case 3, since there will be only one xxhash link statement.